src/serac/numerics/functional/functional.hpp

```diff
@@ -384,14 +390,14 @@ class Functional<test(trial), exec> : public mfem::Operator {
         auto& K_belem = form_.bdr_element_gradients_;
         auto& LUT     = lookup_tables.bdr_element_nonzero_LUT;
 
-        zero_out(K_belem);
+        detail::zero_out(K_belem);
         for (auto& boundary : form_.bdr_integrals_) {
-          boundary.ComputeElementGradients(view(K_belem));
+          boundary.ComputeElementGradients(K_belem);
         }
 
-        for (size_t e = 0; e < K_belem.size(0); e++) {
-          for (size_t i = 0; i < K_belem.size(1); i++) {
-            for (size_t j = 0; j < K_belem.size(2); j++) {
+        for (axom::IndexType e = 0; e < K_belem.shape()[0]; e++) {
+          for (axom::IndexType i = 0; i < K_belem.shape()[1]; i++) {
+            for (axom::IndexType j = 0; j < K_belem.shape()[2]; j++) {
```

Comment:
this is beyond the scope of this PR, but given that this pattern shows up a lot, it might be worth defining an iterator option:

```cpp
for (auto i : K_belem.range(0)) {
  ...
}
```

Reply:
Added to https://github.com/LLNL/axom/issues/633 for tracking purposes